src/bp/ui-studio/src/web/views/FlowBuilder/nodeProps/ActionDialog.tsx

```diff
@@ -0,0 +1,167 @@
+}
+
+interface ActionDialogProps {
+  name: string
+  parameters: Parameters
+  actionServerId: string
+  isOpen: boolean
+  onClose: () => void
+  onSave: (action: Action) => void
+}
+
+const ActionDialog: FC<ActionDialogProps> = props => {
+  const { isOpen, onClose, onSave } = props
+
+  const [actionServers, setActionServers] = useState<ActionServersWithActions[]>([])
+  const [name, setName] = useState(props.name)
+  const [parameters, setParameters] = useState(props.parameters)
+  const [actionServerId, setActionServerId] = useState(props.actionServerId)
+
+  useEffect(() => {
+    const fetchActionServers = async () => {
+      const response = await axios.get(`${window.BOT_API_PATH}/actionServers`)
+      setActionServers(response.data)
+    }
+
+    // tslint:disable-next-line: no-floating-promises
+    fetchActionServers()
+  }, [])
+
+  if (actionServers.length === 0) {
+    return null
+  }
+
+  const actionIsValid = () => {
+    return !!name && !!actionServerId
+  }
+
+  const isNewNode = name === '' && _.isEmpty(parameters) && actionServerId === ''
+  if (isNewNode) {
+    const defaultActionServer = actionServers[0]
+    const defaultAction = defaultActionServer.actions[0]
+
+    const defaultName = defaultAction?.name
+    const defaultParameters = {}
+    const defaultActionServerId = defaultActionServer.id
+
+    setName(defaultName)
+    setParameters(defaultParameters)
+    setActionServerId(defaultActionServerId)
+    if (actionIsValid()) {
+      onSave({ name: defaultName, parameters: defaultParameters, actionServerId: defaultActionServerId })
+    }
+    return null
+  }
+  const currentActionServer = actionServers.find(s => s.id === actionServerId)
+  const currentActionDefinition =
+    currentActionServer.actions.find(a => a.name === name) || currentActionServer.actions[0]
+
+  return (
+    <Dialog isOpen={isOpen} title="Edit Action" icon="offline" onClose={() => onClose()}>
+      <div
+        className={style.actionDialogContent}
+        onMouseDown={e => {
+          // TODO: check for a more elegant way to stop event propagation
+          e.stopPropagation()
+        }}
+        onContextMenu={e => {
+          // TODO: check for a more elegant way to stop event propagation
+          e.stopPropagation()
+        }}
+      >
+        <Label>
+          Action Server
+          <HTMLSelect
+            value={actionServerId}
+            onChange={e => {
+              e.preventDefault()
+              const actionServerId = e.target.value
+              setActionServerId(actionServerId)
+              const actionServer = actionServers.find(s => s.id === actionServerId)
+              setName(actionServer.actions[0].name)
+            }}
+          >
+            {actionServers.map(actionServer => (
+              <option key={actionServer.id} value={actionServer.id}>
+                {actionServer.id} ({actionServer.baseUrl})
+              </option>
+            ))}
+          </HTMLSelect>
+        </Label>
```

Comment:
I've created a small dropdown component in the admin UI (could be copied to ui-shared) which is really simple to use instead of handling manually an HTML Select.

Usage: https://github.com/botpress/botpress/blob/master/src/bp/ui-admin/src/Pages/Logs/index.tsx#L148
Component: https://github.com/botpress/botpress/blob/master/src/bp/ui-admin/src/Pages/Components/Dropdown.tsx